load.js

```diff
@@ -4,39 +4,60 @@ var globby = require('globby');
 var path = require('path');
 var protobuf = require('protobufjs');
 
-var COMMON_PROTO_DIRS = [
-  // This list of directories is defined here:
-  // https://github.com/googleapis/googleapis/blob/master/gapic/packaging/common_protos.yaml
-  'api',
-  path.join('logging', 'type'),
-  'longrunning',
-  'protobuf', // This is an additional path that the common protos depend on.
-  'rpc',
-  'type',
-];
-
-var COMMON_PROTO_GLOB_PATTERNS = COMMON_PROTO_DIRS.map(function(dir) {
-  return path.join('google', dir, '**', '*.proto');
-});
-
-var COMMON_PROTO_FILES = globby.sync(COMMON_PROTO_GLOB_PATTERNS);
+let COMMON_PROTO_FILES;
 
 class GoogleProtoFilesRoot extends protobuf.Root {
   constructor() {
     super([].slice.apply(arguments));
   }
 
+  // Caches and returns an array of the local common/google core proto files
+  // exist on disk.
+  static getCommonProtoFiles() {
+    var commonProtoDirs = [
+      // This list of directories is defined here:
+      // https://github.com/googleapis/googleapis/blob/master/gapic/packaging/common_protos.yaml
+      'api',
+      path.join('logging', 'type'),
+      'longrunning',
+      'protobuf', // This is an additional path that the common protos depend on.
+      'rpc',
+      'type',
+    ];
+
+    var commonProtoGlobPatterns = commonProtoDirs.map(dir =>
+      path.join(__dirname, 'google', dir, '**', '*.proto')
+    );
+
+    if (!COMMON_PROTO_FILES) {
+      COMMON_PROTO_FILES = globby
+        .sync(commonProtoGlobPatterns)
+        .map(path.normalize);
+    }
+
+    return COMMON_PROTO_FILES;
+  }
+
   // Causes the loading of an included proto to check if it is a common
   // proto. If it is a common proto, use the google-proto-files proto.
   resolvePath(_, includePath) {
+    includePath = path.normalize(includePath);
```

Comment:
is this just for safety? You also normalize when populating COMMON_PROTO_FILES

Reply:
Yeah, globby actually reverses the slashes, and we don't know what's coming into `resolvePath`, so this just makes sure there's a consistent format between the input and the list we made of files on disk.